File: C++/1886. Determine Whether Matrix Can Be Obtained By Rotation.cpp
class Solution {
public:
    void vo(vector<vector<int>>& m){
        for(int i=0;i<m.size();i++){
            for(int j=i+1;j<m.size();j++)
            swap(m[i][j],m[j][i]);
        }
        for(int i=0;i<m.size();i++){
                reverse(m[i].begin(),m[i].end());
        }
    }
    bool findRotation(vector<vector<int>>& m, vector<vector<int>>& t) {
        for(int i=0;i<4;i++){
            if(m==t)
            return true;
            vo(m);
        }
        return false;
    }
};

File: JAVA/1886. Determine Whether Matrix Can Be Obtained By Rotation.java
class Solution {
    public void vo(int[][] m){
        for(int i=0;i<m.length;i++){
            for(int j=i+1;j<m.length;j++){
                int t=m[i][j];
                m[i][j]=m[j][i];
                m[j][i]=t;
            }
        }
        for(int i=0;i<m.length;i++){
            for(int j=0;j<m.length/2;j++){
                int t=m[i][j];
                m[i][j]=m[i][m.length-1-j];
                m[i][m.length-1-j]=t;
            }
        }
    }
    public boolean findRotation(int[][] m, int[][] t) {
        for(int i=0;i<4;i++){
            if(Arrays.deepEquals(m,t))
            return true;
            vo(m);
        }
        return false;
    }
    
}